kubernetes/host-logs-geneva.yaml

```diff
@@ -0,0 +1,96 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+ name: host-logs-windows
+ namespace: kube-system
+ labels:
+  component: geneva-agent-windows
+  tier: node-win
+spec:
+ updateStrategy:
+  type: RollingUpdate
+ selector:
+  matchLabels:
+    component: geneva-agent-windows
+    tier: node-win
+ template:
+  metadata:
+    labels:
+      component: geneva-agent-windows
+      tier: node-win
+    annotations:
+      agentVersion: "0.0.0-0"
+      dockerProviderVersion: "18.0.1-0"
+      schema-versions: "v1"
+  spec:
+    dnsConfig:
+      options:
+        - name: ndots
+          value: "3"
+    securityContext:
+      windowsOptions:
+        hostProcess: true
+        runAsUserName: "NT AUTHORITY\\SYSTEM"
+    hostNetwork: true
+    containers:
+     - name: host-logs-windows
+       image: "VALUE_IMAGE_NAME"
+       imagePullPolicy: IfNotPresent
+       resources:
+        limits:
+         cpu: 500m
+         memory: 600Mi
+       env:
+        # azure devops pipeline uses AKS_RESOURCE_ID and AKS_REGION hence ensure to uncomment these
+        - name: AKS_RESOURCE_ID
+          value: "VALUE_AKS_RESOURCE_ID_VALUE"
+        - name: AKS_REGION
+          value: "VALUE_AKS_RESOURCE_REGION_VALUE"
+        - name: CONTROLLER_TYPE
+          value: "DaemonSet"
+        - name: HOSTNAME
+          valueFrom:
+             fieldRef:
+               fieldPath: spec.nodeName
+        - name: PODNAME
+          valueFrom:
+             fieldRef:
+              fieldPath: metadata.name
+        - name: NODE_IP
+          valueFrom:
+             fieldRef:
+               fieldPath: status.hostIP
+       volumeMounts:
+        - mountPath: \etc\config\settings
+          name: settings-vol-config
+          readOnly: true
+       livenessProbe:
+          exec:
+            command:
+              - cmd
+              - /c
+              - .\opt\hostlogswindows\scripts\cmd\livenessprobe.exe
```

Comment:
Missing the genevagent exe here. Liveness probe should monitor the geneva agent process. If the process terminates/hangs then liveness probe should trigger the restart.

Reply:
Should this be put here instead of main.ps1

`if(Get-GenevaEnabled){
    Invoke-Expression ".\opt\genevamonitoringagent\genevamonitoringagent\Monitoring\Agent\MonAgentLauncher.exe -useenv"
} else {
    Write-Host "Geneva not configured. Watching for config map"
    # Infinite loop keeps container alive while waiting for config map
    # Otherwise when the process ends, kubernetes sees this as a crash and the container will enter a crash loop
    while($true){
        Start-Sleep 5
    }
}`

Reply:
It might make sense to check Get-Geneva enabled in both places. We don't want to start the Geneva agent without the GCS variables, since we know it will crash. But we also don't want the liveness probe to trigger a restart if we intentionally didn't start the agent while waiting for a configmap to be mounted. If we're not consistent, we could get an infinite loop of false negatives from the liveness probe whenever the configmap isn't mounted yet

Reply:
We have deiced to make this specific instance a separate PR and discussion will conduct during our next weekly meeting with @ganga1980. 

As the image stays separate from Container Insights main offering this isn't a must for us but will still be considered in scope until we can confirm with Ganga.